File: src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitPhoto/EngineerVisitPhoto.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace EngineerVisitPhotos

Public Class EngineerVisitPhoto

    Private _dataTypeValidator As DataTypeValidator
    Public Sub New()
        _dataTypeValidator = New DataTypeValidator
    End Sub

#Region "Class Properties"

    Public Property IgnoreExceptionsOnSetMethods() As Boolean
        Get
            Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
        End Get
        Set(ByVal Value As Boolean)
            Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
        End Set
    End Property

    Public ReadOnly Property Errors() As Hashtable
        Get
            Return _dataTypeValidator.Errors
        End Get
    End Property

    Public ReadOnly Property DTValidator() As DataTypeValidator
        Get
            Return _dataTypeValidator
        End Get
    End Property
    
    Private _exists As Boolean = False
    Public Property Exists() As Boolean
        Get
            Return _exists
        End Get
        Set(ByVal Value As Boolean)
            _exists = Value
        End Set
    End Property

    Private _deleted As Boolean = False
    Public ReadOnly Property Deleted() As Boolean
        Get
            Return _deleted
        End Get
    End Property
    Public WriteOnly Property SetDeleted() As Boolean
        Set(ByVal Value As Boolean)
            _deleted = Value
        End Set
    End Property
    
#End Region

#Region "EngineerVisitPhoto Properties"

      
Private _EngineerVisitPhotoID As Integer = 0
Public Readonly Property EngineerVisitPhotoID() As Integer
	Get 
		Return _EngineerVisitPhotoID
	End Get	
End Property	
Public Writeonly Property SetEngineerVisitPhotoID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_EngineerVisitPhotoID", Value)
	End Set
End Property


Private _EngineerVisitID As Integer = 0
Public Readonly Property EngineerVisitID() As Integer
	Get 
		Return _EngineerVisitID
	End Get	
End Property	
Public Writeonly Property SetEngineerVisitID() As Object
	Set(Value as Object)
		_dataTypeValidator.SetValue(Me, "_EngineerVisitID", Value)
	End Set
End Property

            Private _Photo As Byte() = Nothing
            Public ReadOnly Property Photo() As Byte()
                Get
                    Return _Photo
                End Get
            End Property
            Public WriteOnly Property SetPhoto() As Object
                Set(ByVal Value As Object)
                    _Photo = Value
                End Set
            End Property

            Private _Caption As String = String.Empty
            Public ReadOnly Property Caption() As String
                Get
                    Return _Caption
                End Get
            End Property
            Public WriteOnly Property SetCaption() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Caption", Value)
                End Set
            End Property



#End Region

End Class

End Namespace

End Namespace

File: apollo/apollo.fsm/src/Application Specific/Classes/Entity Collections/Jobs/EngineerVisit/EngineerVisitPhoto/EngineerVisitPhoto.cs
using System.Collections;

namespace FSM.Entity
{
    namespace EngineerVisitPhotos
    {
        public class EngineerVisitPhoto
        {
            private DataTypeValidator _dataTypeValidator;

            public EngineerVisitPhoto()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            
            

            private int _EngineerVisitPhotoID = 0;

            public int EngineerVisitPhotoID
            {
                get
                {
                    return _EngineerVisitPhotoID;
                }
            }

            public object SetEngineerVisitPhotoID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_EngineerVisitPhotoID", value);
                }
            }

            private int _EngineerVisitID = 0;

            public int EngineerVisitID
            {
                get
                {
                    return _EngineerVisitID;
                }
            }

            public object SetEngineerVisitID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_EngineerVisitID", value);
                }
            }

            private byte[] _Photo = null;

            public byte[] Photo
            {
                get
                {
                    return _Photo;
                }
            }

            public object SetPhoto
            {
                set
                {
                    _Photo = (byte[])value;
                }
            }

            private string _Caption = string.Empty;

            public string Caption
            {
                get
                {
                    return _Caption;
                }
            }

            public object SetCaption
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Caption", value);
                }
            }



            
        }
    }
}